src/include/utime.h

```diff
@@ -18,6 +18,15 @@
 #include <math.h>
 #include <sys/time.h>
 #include <time.h>
+
+#if defined(DARWIN)
+/* For some reason this definition is lost when compiling ceph *
+ * It works fine on a stand-alone file even with same clang    *
+ * parameters                                                  */
+
+time_t timegm(struct tm *);
```

Comment:
@schafdog seems we can dispense with this change as i am able to compile the tree without it.

and i checked "time.h" offered by apple. we should have `timegm()` declared if we include "time.h".

and we don't define `_ANSI_SOURCE` or `_POSIX_C_SOURCE` (i 

- put `#ifdefs #warning` in "test_crc32c.cc" to print messages if any of them are defined
- `make test/rgw/rgw_json_enc.o` without this change

to verify my theory).